setup/gcp/installer/forseti_installer.py

```diff
@@ -0,0 +1,291 @@
+
+
+class ForsetiInstaller:
+    """Forseti installer base class (abstract)"""
+    __metaclass__ = ABCMeta
+
+    # Class variables initialization
+    branch = None
+    project_id = None
+    organization_id = None
+    gcp_service_account = None
+    user_can_grant_roles = True
+    config = Config()
+
+    @abstractmethod
+    def __init__(self):
+        """Initialize."""
+        pass
+
+    def run_setup(self):
+        """Run the setup steps"""
+        print_banner('Installing Forseti {} v{}'.format(
+            self.config.template_type, get_forseti_version()))
+
+        # Preflight checks
+        self.preflight_checks()
+
+        # Deployment
+        bucket_name = self.generate_bucket_name()
+        conf_file_path = self.generate_forseti_conf()
+        deployment_tpl_path = self.generate_deployment_templates()
+
+        deploy_success, deployment_name = self.deploy(deployment_tpl_path,
+                                                      conf_file_path,
+                                                      bucket_name)
+
+        # After deployment
+        self.post_install_instructions(deploy_success,
+                                       deployment_name,
+                                       deployment_tpl_path,
+                                       conf_file_path,
+                                       bucket_name)
+
+    def preflight_checks(self):
+        """Pre-flight checks"""
+        self.check_run_properties()
+        self.branch = infer_version(self.config.advanced_mode)
+        self.project_id, authed_user, is_devshell = get_gcloud_info()
+        verify_gcloud_information(self.project_id,
+                                  authed_user,
+                                  self.config.force_no_cloudshell,
+                                  is_devshell)
+        self.organization_id = lookup_organization(self.project_id)
+        check_billing_enabled(self.project_id, self.organization_id)
+        self.format_gcp_service_acct_id()
+        self.gcp_service_account = create_reuse_service_acct(
+            'gcp_service_account',
+            self.gcp_service_account,
+            self.config.advanced_mode,
+            self.config.dry_run)
+
+    def deploy(self, deploy_tpl_path, conf_file_path, bucket_name):
+        """Deploy Forseti using the deployment template
+
+        Args:
+            deploy_tpl_path (str): Deployment template path
+            conf_file_path (str): Configuration file path
+            bucket_name (str): Name of the GCS bucket
+
+        Returns:
+            bool: Whether or not the deployment was successful
+            str: Deployment name
+        """
+        deployment_name, return_code = create_deployment(
+            self.project_id,
+            self.organization_id,
+            deploy_tpl_path,
+            self.config.template_type,
+            self.config.datetimestamp,
+            self.config.dry_run)
+
+        if not return_code:
```

Comment:
Nit, I recommend making a new variable instead of using 'not return_code'.

deployment_completed = not return_code
if deployment_completed:
...
return deployment_completed, deployment_name

Reply:
Updated.